### ISSUE TITLE
Request: Support ABBA, a new LoRA type that can often outperform finetuning

### ISSUE THREAD

**Issue by araleza** [kohya-ss/sd-scripts#2123]
This week, a new paper announced ABBA, a variation of LoRA that the authors claim has been "consistently outperforming existing PEFT
methods by a significant margin across multiple models":

https://arxiv.org/pdf/2505.14238

It looks simple to implement if you already know LoRA. Instead of two matrices A and B being multiplied together like LoRA has for each target matrix, it multiplies A and B, then another independent A and B matrix, and then Hadamard multiplies these two results together. The two AB matrices instead of a single set are why the technique has been named ABBA.

The authors note that this technique sometimes outperforms a full fine tune. That's possibly because the Hadamard multiplication is producing the correct 'shape' of change across multiple weights that individual weights' gradients don't capture so easily.

Although it hasn't been tested on image generation networks yet, there's a reasonable chance it'll perform just as well.

In fact, in practice for local Flux training, it might regularly perform even better than full fine tuning. That's because even on a 5090 GPU, there's not enough VRAM to allow training to use the Adam optimizer. Only the Adafactor optimizer works right now, because even though it's significantly weaker at optimizing than Adam, it uses much less memory, allowing it to fit on local consumer GPUs. So if we had a variant of LoRA available that could even approximately match full fine tuning, then being able to use Adam for it would improve its chances of outperforming a full fine tune that can only use Adafactor.

**Comment by deGENERATIVE-SQUAD** [comment]
@araleza 
@KohakuBlueleaf said that it's LoHA algo from LyCORIS lib

**Comment by KohakuBlueleaf** [comment]
> [@araleza](https://github.com/araleza) [@KohakuBlueleaf](https://github.com/KohakuBlueleaf) said that it's LoHA algo from LyCORIS lib

<img width="776" height="347" alt="Image" src="https://github.com/user-attachments/assets/260711d1-eb13-4dd9-a8e7-7c45583c1266" />

<img width="658" height="540" alt="Image" src="https://github.com/user-attachments/assets/f1fe31fc-847a-4345-b2f6-79512691da5f" />

Exactly Same, they just use SVD to initialize the weight.

**Comment by araleza** [comment]
Yeah I think you're right. I've been reading a discussion between the paper author and various people on adding ABBA to the huggingface peft library, and they seem to have decided that the implementation is just LoHa but with

* Different weight initialization (but this doesn't seem to be a game changer in terms of results)
* More optimal code implementation due to Khatri-Rao

The discussion is here:

https://github.com/huggingface/peft/issues/2587
